### PR TITLE
Initial Jlink support

### PR DESCRIPTION
**Pull request by ysb33r**
- Add conditional composite build on newrelic-telemetry-sdk-java
- Prefer plugin versions to be defined in root project
- Use JDK14
- Remove dependency on shaded Gson
- Add JLink build
- Removed use of Shadow jar
- Builds normal `tar.bz2`,` zip`, then jlinked versions of `tar.bz2`, `zip` and finally also `rpm` and `deb`.
(No current support for building Windows MSIs).

### CODE REVIEW

**Review by kittylyst — APPROVED** [review]
LGTM.

File: jfr-daemon/build.gradle.kts [hunk 1]
```diff
@@ -15,26 +16,20 @@ java {
...
-    implementation("com.google.code.gson:gson:${Versions.gson}")
-}
-
-tasks.shadowJar {
-    archiveClassifier.set("")
-    manifest {
-        attributes(
-                "Main-Class" to "com.newrelic.jfr.daemon.JFRDaemon",
-                "Implementation-Version" to project.version
-        )
```
Comment (jack-berg) [inline jfr-daemon/build.gradle.kts:33]: So this changes the behavior of the daemon in a way that isn't documented. The project still publishes a jar, but its no longer a fat jar and doesn't function as described here: https://docs.newrelic.com/docs/agents/java-agent/features/real-time-profiling-java-using-jfr-metrics

The publish task will still run, but its just a regular jar so its functionally broken.

File: jfr-daemon/build.gradle.kts [hunk 2]
```diff
@@ -15,26 +16,20 @@ java {
...
+    }
+    module("objenesis-${objenesisVersion}.jar", "org.objenesis", objenesisVersion) {
+        exports("org.objenesis")
     }
 }
 
-tasks.named("build") { dependsOn("shadowJar") }
+dependencies {
+    implementation(project(":jfr-mappers"))
+    implementation(jpms.asModule("org.apache.logging.log4j:log4j-core:${log4jVersion}", "org.apache.logging.log4j.core"))
```
Comment (jack-berg) [inline jfr-daemon/build.gradle.kts:30]: I've tried to create and run the jlink artifacts, and I see a variety of thinkgs in the `jfr-daemon/build/jlink` directory, but when I run them its not clear whether they're working or not. I think there may be a problem with the logging. 

I notice that the slf4j dependencies are missing and wondering if that has something to do with it. What's the expected behavior of logging with jlink artifacts?

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+}
+
 rootProject.name = "newrelic-jfr-core"
 
 include("jfr-mappers")
 include("jfr-daemon")
+
+//if(file("../newrelic-telemetry-sdk-java").exists()){
+//    includeBuild("../newrelic-telemetry-sdk-java")
+//}
```
Comment (jack-berg) [inline settings.gradle.kts:17]:
```suggestion
```

File: settings.gradle.kts [hunk 4]
```diff
@@ -1,4 +1,17 @@
+pluginManagement {
+    repositories {
+//        mavenLocal() // ***** REMOVE BEFORE MERGE
```
Comment (jack-berg) [inline settings.gradle.kts:3]:
```suggestion
```

File: jfr-mappers/build.gradle.kts [hunk 5]
```diff
@@ -13,6 +18,23 @@ tasks {
...
+    module("gson-${gsonVersion}.jar", "com.google.code.gson", gsonVersion) {
+        exports("com.google.gson")
+    }
+    module("mockito-junit-jupiter-${mockitoVersion}.jar", "mockito.junit.jupiter", mockitoVersion) {
+        exports("com.google.gson")
+    }
+    module("objenesis-${objenesisVersion}.jar", "org.objenesis", objenesisVersion) {
+        exports("com.google.gson")
+    }
+}
```
Comment (jack-berg) [inline jfr-mappers/build.gradle.kts:36]: I believe `jpmsExtraModules` is coming from the `org.ysb33r.java.modulehelper` plugin, but its not clear to me what that plugin is doing. Can you elaborate on what happening here?

File: .sdkmanrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Enable auto-env through the sdkman_auto_env config
+# Add key=value pairs of SDKs to use below
+java= 11.0.8.hs-adpt
```
Comment (jack-berg) [inline .sdkmanrc:3]: I don't think we should have sdkman specifics checked into this project.

**Comment by jack-berg** [comment]
Closing this PR because this code was merged [here](https://github.com/newrelic/newrelic-jfr-core/pull/128)

**Pull request closed without merging**